src/main/java/uk/ac/ebi/eva/pipeline/io/readers/NonAnnotatedVariantsMongoReader.java

```diff
@@ -31,13 +33,22 @@
  */
 public class NonAnnotatedVariantsMongoReader extends MongoDbCursorItemReader {
 
-    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName) {
-        super();
+    private static final String STUDY_KEY = VariantToDBObjectConverter.FILES_FIELD + "."
+            + VariantSourceEntryToDBObjectConverter.STUDYID_FIELD;
 
+    /**
+     * @param studyId nullable. If null, the query brings all non-annotated variants in the collection.
+     *  If not null, bring only non-annotated variants from that study.
+     */
+    public NonAnnotatedVariantsMongoReader(MongoOperations template, String collectionsVariantsName, String studyId) {
         setTemplate(template);
         setCollection(collectionsVariantsName);
 
-        DBObject query = BasicDBObjectBuilder.start().add("annot.ct.so", new BasicDBObject("$exists", false)).get();
+        BasicDBObjectBuilder queryBuilder = BasicDBObjectBuilder.start();
+        if (!studyId.isEmpty()) {
```

Comment:
This will break if `studyId` is null. Please also add the corresponding test case.